docs/platforms/javascript/common/enriching-events/transaction-name/index.mdx

```diff
@@ -22,15 +40,104 @@ task being executed. For example:
 Ideally, the transaction name does not contain variable values such as user
 IDs but has rather low cardinality while still uniquely identifying a piece of
 code you care about.
+For example, instead of `'GET /api/users/123/details'`, set `'GET /api/users/:id/details'`.
+This ensures that errors or traces for all requests made to this route are grouped together.
+
+### When to set the Transaction Name
+
+We generally recommend to let the SDK set the transaction name automatically.
+However, in some cases, the SDK might not be able to set a name or the set name doesn't work for your use case.
+
+If you need to set the transaction name manually, we recommend setting it _as early as possible_ in your application code, specifically when the operation you want to describe with the transaction name starts.
+This ensures that errors thrown early in the life cycle of your application are correctly annotated.
+Furthermore, it ensures consistency across a trace if you use [distributed tracing](/concepts/key-terms/tracing/distributed-tracing/).
+
+For example, in a server application, you might set the transaction name in a middleware that runs before the request is processed.
+In a web application, you might set the transaction name when you enter a new route (e.g. in an SPA Router).
+
+## Setting the Error Transaction Name
+
+There are several ways to set the transaction name for error events sent to Sentry.
```

Comment:
In "Setting the Root Span Name", you mention where this name is displayed and used for. You could include this information here as well.